### PR TITLE
Adding in Truffle examples per discussion

### PR DESCRIPTION
**Pull request by gskapka**
Note that I've made the truffle example a _stand alone_, in that it has a copy of the oraclize API & URL requests contract inside it. Could work out something to symlink it all together at a later date, but for now this is an MVP of what it could look like. 

Note: The image linked in the README won't display until it's in the main repo, and just sits there as a link that hits a 404 right now (despite copying the link whilst viewing the image in a browswer!)

### CODE REVIEW

File: solidity/truffle-examples/url-requests/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "Oraclize Examples using Truffle: Computation-Datasource - Url Requests",
```
Comment (D-Nice) [inline solidity/truffle-examples/url-requests/package.json]: npm will complain but silently fail due to this invalid name. Same with pnpm (it throws errors in fact). https://www.npmjs.com/package/validate-npm-package-name

File: solidity/truffle-examples/url-requests/test/url-requests-tests.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
+const Web3 = require('web3')
+const waitForEvent = require('./utils')
```
Comment (D-Nice) [inline solidity/truffle-examples/url-requests/test/url-requests-tests.js]: all tests dependent on it fail with:
```
  4) Contract: Oraclize Example using Truffle
       URL Requests Tests
         Should emit result from PUT request:
     TypeError: _event is not a function
      at Promise (test/utils.js:3:5)
      at new Promise (<anonymous>)
      at module.exports (test/utils.js:2:3)
      at Context.it (test/url-requests-tests.js:136:45)
      at process.topLevelDomainCallback (domain.js:121:23)

  5) Contract: Oraclize Example using Truffle
       URL Requests Tests
         Should emit result from cookie request:
     TypeError: _event is not a function
      at Promise (test/utils.js:3:5)
      at new Promise (<anonymous>)
      at module.exports (test/utils.js:2:3)
      at Context.it (test/url-requests-tests.js:165:45)
      at process.topLevelDomainCallback (domain.js:121:23)
```

File: solidity/truffle-examples/url-requests/package.json (hunk 1 above)
Comment (gskapka) [reply solidity/truffle-examples/url-requests/package.json]: TIL! 

Have renamed to the following: `"oraclize-examples-using-truffle__computation-datasource-url-requests"`

Suitable?

**Comment by D-Nice** [comment]
Looking good now and passing for most part. 

There were instances where the following test would return an error... Of course when I added additional logging to find the culprit, it would pass:

```
  1) Contract: Oraclize Example using Truffle                                                                                                                               
       URL Requests Tests                                                                                                                                                        
         Should emit result from POST request:                                                                                                                                                                                                             
                                                                                                                                                                                                                        
      Incorrect result from POST request!                                                                                                                                   
      + expected - actual                                                                                                                                                                                                                                  
                                                                                                                                                                            
      -{"status":200,"result":[{"query":"OX49 5NU","result":{"postcode":"OX49 5NU","quality"                                                                                
                                                                                                                                                                                                                                                           
      at Context.it (test/url-requests-tests.js:132:14)                                                                                                                     
      at process._tickCallback (internal/process/next_tick.js:68:7)

**Pull request merged**